src/browser.ts

```diff
@@ -96,7 +104,8 @@ export class Browser {
       url: "",
     });
     const browserWsUrl = new URL(this.#celestial.ws.url);
-    const wsUrl = `${browserWsUrl.origin}/devtools/page/${targetId}`;
+    const wsUrl =
+      `${browserWsUrl.origin}/devtools/page/${targetId}${browserWsUrl.search}`;
```

Comment:
The `browserWsUrl.search` is required because some service may use this for authentication
For example browserless.io:
```ts
  const browser = await puppeteer.connect({
    browserWSEndpoint: `wss://chrome.browserless.io?token=******`,
  });